selfservice/hook/web_hook.go

```diff
@@ -209,7 +210,15 @@ func (e *WebHook) ExecutePostRegistrationPostPersistHook(_ http.ResponseWriter,
 	if gjson.GetBytes(e.conf, "can_interrupt").Bool() || gjson.GetBytes(e.conf, "response.parse").Bool() {
 		return nil
 	}
-	return otelx.WithSpan(req.Context(), "selfservice.hook.WebHook.ExecutePostRegistrationPostPersistHook", func(ctx context.Context) error {
+
+	// We want to decouple the request from the hook execution, so that the hooks still execute even
+	// if the request is canceled.
```

Comment:
I don't think we actually want this, do we?

Reply:
I think we do. Because otherwise the user can decide (through cancelling the request at the appropriate time) whether or not a post persist hook is executed. And if the post-hook is cancelled, the user is still persisted. That would mean that our customers can't put any faith in the execution of the post-persist hooks.

Reply:
IMO not for every hook, but only for the "async", non-blocking hooks. The sync hooks need to share context with the request.

Reply:
> I think we do. Because otherwise the user can decide (through cancelling the request at the appropriate time) whether or not a post persist hook is executed. And if the post-hook is cancelled, the user is still persisted. That would mean that our customers can't put any faith in the execution of the post-persist hooks.

That actually makes a lot of sense to me.